core/src/main/scala/io/finch/Input.scala

```diff
@@ -10,16 +10,21 @@ import org.jboss.netty.handler.codec.http.multipart.{DefaultHttpDataFactory, Htt
 import shapeless.Witness
 
 /**
- * An input for [[Endpoint]].
+ * An input for [[Endpoint]] that glues two individual pieces together:
+ *
+ * - Finagle's [[Request]] needed for evaluating (e.g., `body`, `param`)
+ * - Finch's route (represented as `Seq[String]`) needed for matching (e.g., `path`)
  */
-final case class Input(request: Request, path: Seq[String]) {
-  def headOption: Option[String] = path.headOption
```

Comment:
Is the idea here that we can just call these directly on `route` and duplicating methods from `Seq` was unnecessary?